vlm/src/main/scala/geotrellis/contrib/vlm/RasterSourceRDD.scala

```diff
@@ -114,6 +114,39 @@ object RasterSourceRDD {
     ContextRDD(result, layerMetadata)
   }
 
+  def apply(
+    sources: RDD[RasterSource],
+    layout: LayoutDefinition
+  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] = {
+    val cellType: CellType = sources.map { _.cellType }.reduce { _ union _ }
+
+    val projections: Set[CRS] = sources.map { source => Set(source.crs) }.reduce { _ ++ _ }
+    require(
+      projections.size == 1,
+      s"All RasterSources must be in the same projection, but multiple ones were found: $projections"
+    )
+
+    val crs = projections.head
+
+    val mapTransform = layout.mapTransform
+    val extent = mapTransform.extent
+    val combinedExtents = sources.map { _.extent }.reduce { _ combine _ }
+
+    val layerKeyBounds = KeyBounds(mapTransform(combinedExtents))
+
+    val layerMetadata =
+      TileLayerMetadata[SpatialKey](cellType, layout, combinedExtents, crs, layerKeyBounds)
+
+    val keyedRDD: RDD[(SpatialKey, MultibandTile)] =
+      sources.flatMap { case source =>
+        val layoutTileSource = LayoutTileSource(source, layout)
+
+        layoutTileSource.readAll(layoutTileSource.keys.toIterator)
```

Comment:
Look here to find an example of `RDD[RasterSource]` conversion into `TileLayerRDD`:

https://github.com/geotrellis/geotrellis-contrib/blob/master/vlm/src/test/scala/geotrellis/contrib/vlm/RasterSummarySpec.scala#L96-L103

In this code you're loading the entire `RasterSource` into a single partition. Will definitely be a bottleneck.

Reply:
Oh, I completely missed that. Thank you for pointing that out.